module-4/hashcat/README.md

```diff
@@ -0,0 +1,95 @@
+# My `hashcat` tool
+
+This folder contains the source code for my own implementation of the `hashcat` tool.
+Hashcat can be used to TBD.
+
+This is a lightweight implementation that covers the following features/options:
+
+- `-m` `--mode`, Hash mode: 0 (MD5), 1 (SHA-1), 2 (SHA-256), 3 (SHA-512)
+- `-a` `--attack`, Attack mode: 0 (Brute-Force), 1 (Dictionary)
+- `-h` `--hash`, Target hash (required if -H is not used)
+- `-H` `--hashfile`, File containing the target hash (required if -h is not used)
+- `-w` `--wordlist`,Path to the wordlist file (required for Dictionary attack)
+- `--help`, Show this help message and exit
+
+## Features
+
+- **Brute Force Attack**: Attempts all possible combinations of a specified character set within given length constraints to find the plaintext password.
+- **Dictionary Attack**: Uses a provided wordlist to attempt to find the plaintext password for the target hash.
+- **Multiple Hash Algorithms**: Supports MD5, SHA-1, SHA-256, and SHA-512.
+- **Command-Line Interface**: Allows flexible configuration of attack parameters through command-line arguments.
+
+## Getting started
+
+### Prerequisites
+
+- Python 3.6 or later
+
+### Installation
+
+1. Clone or download the script.
+2. Ensure you have Python installed.
```

Comment:
make this more instructional. 
this document serves as a starting point and as a user, I want to just follow your instructions as a developer to make your tool work for me, too